src/AppBundle/Controller/BlogController.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace AppBundle\Controller;
+
+use Symfony\Bundle\TwigBundle\TwigEngine;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use eZ\Publish\API\Repository\SearchService;
+use AppBundle\QueryType\ChildrenQueryType;
+
+class BlogController
+{
+    /** @var \Symfony\Bundle\TwigBundle\TwigEngine */
+    private $templating;
+
+    /** @var \eZ\Publish\API\Repository\SearchService */
+    private $searchService;
+
+    /** @var \AppBundle\QueryType\ChildrenQueryType */
+    private $childrenQueryType;
+
+    /** @var int */
+    private $blogLocationId;
+
+    /** @var int */
+    private $blogPostsLimit;
+
+    /**
+     * @param \Symfony\Bundle\TwigBundle\TwigEngine $templating
+     * @param \eZ\Publish\API\Repository\SearchService $searchService
+     * @param \AppBundle\QueryType\ChildrenQueryType $childrenQueryType
+     * @param int $blogLocationId
+     * @param int $blogPostsLimit
+     */
+    public function __construct(
+        TwigEngine $templating,
```

Comment:
could we use `Symfony\Component\Templating\EngineInterface` interface intead of `TwigEngine` class?